config/config.go

```diff
@@ -165,6 +166,26 @@ func (t *GDPRTimeouts) ActiveTimeout() time.Duration {
 	return time.Duration(t.ActiveVendorlistFetch) * time.Millisecond
 }
 
+// TCF2 defines the TCF2 specific configurations for GDPR
+type TCF2 struct {
+	Enabled             bool                 `mapstructure:"enabled"`
+	Purpose1            PurposeDetail        `mapstructure:"purpose1"`
+	Purpose2            PurposeDetail        `mapstructure:"purpose2"`
+	Purpose7            PurposeDetail        `mapstructure:"purpose7"`
+	SpecialPurpose1     PurposeDetail        `mapstructure:"special_purpose1"`
```

Comment:
When we add config options for other purposes in the future for the 5/15 spec update, will we add them as Purpose3, etc... or switch to a slice?

Reply:
Is that the cookie sync update where we check for purposes 2-10? I am not sure what the use case would be there for per purpose checks. Currently, if we don't enforce purpose 1, then `allowSync` always passes. When we add in the 2-10 check, what would it mean to not enforce purpose 3 for example? In the `allowSync`, we are not enforcing any purpose other than 1, we just are checking if any one of the other purposes provide a valid reason to want to sync.

Reply:
I had to go back and check. It's new changes to the 'Passing User IDs into an auction' which now states:

Any Purpose 2-10. User IDs are important for more than personalizing ads - they can be used in frequency capping, building profiles, counting unique users, etc. So Prebid Server should pass User IDs through the auction if any of Purposes 2-10 pass the legal basis test.

Would that require expanding the purpose configs? Perhaps not?

Reply:
I think for purpose 2-10, we would just have a list of purposes that the host does not consider a valid reason for PI. That is if the host declares purpose 3 as not valid (on the list), then a vendor who only claims purpose 3 would not get the PI. `Purpose1` and `SpecialPurpose1` will likely stay, 2 and 7 would go away and be replaced by the list. The key change in behavior is we are now looking for "any of" rather than "all" purposes.

Another open question I discussed with Brett after reading the most recent changes in the document is the different types of PI. Geo is already captured with `SpecialPurpose1`, though not explicitly called out that much. User IDs are discussed in the document, but we also have IP addresses, age, and gender data to potentially be scrubbed. Brett said he would bring that up with the group for clarification.

Reply:
Cool. I saw Bret started that discussion. Should the pending results of that discussion hold up this PR?

Reply:
I don't think so ... the only conflict I anticipate is the removal of the `Purpose2` and `Purpose7` sections ... the new stuff should end up in a new section I think